lib/game.rb

```diff
@@ -20,33 +20,45 @@ def toggle_current_player
   end
 
   def make_move
-    @display.display_board(@board.board)
+    @display.display_board(@board.squares)
     @display.show_current_player(@current_player.mark)
     move = @display.ask_for_move
-    until move_valid?(move)
-      @display.invalid_move_message(move)
+    until @board.move_valid?(move)
+      @display.show_invalid_move_message(move)
       move = @display.ask_for_move
     end
     @board.player_make_move(@current_player.mark, move.to_i)
   end
 
-  public
+  def winning_player_exists?
+    @board.has_winning_combination?(@player1.mark) || @board.has_winning_combination?(@player2.mark)
+  end
 
-  def moves_remaining?
-    @board.moves_remaining > 0
+  def winning_player
+    if @board.has_winning_combination?(@player1.mark)
+      @player1.mark
+    else
+      @player2.mark
+    end
   end
 
+  public
+
   def play_move
     make_move
     toggle_current_player
   end
 
-  def can_continue_playing?
-    moves_remaining?
+  def is_over?
+    !@board.moves_remaining? || winning_player_exists?
   end
 
-  def move_valid?(move)
-    move = move.to_i
-    move.between?(1, 9) && @board.position_available?(move)
+  def tie_or_won
```

Comment:
The game is deciding on what message to show to a user. But is also responsible for managing the game itself... This feels like a breach of the SRP.

Reply:
I agree, I'll amend this. Thanks!